_posts/2021-08-24-smartblog.md

```diff
@@ -0,0 +1,82 @@
+---
+layout: post
+title: "[CVE-2021-37538] Improper neutralization of SQL parameter in SmartBlog module from SmartDataSoft for PrestaShop"
+categories: module
+author:
+- Friends-Of-Presta.org
+meta: "CVE,PrestaShop,smartblog"
+severity: "critical (9.8)"
+---
+
+Multiple SQL injection vulnerabilities in SmartDataSoft SmartBlog for PrestaShop before 4.0.6 allow a remote unauthenticated attacker to execute arbitrary SQL commands via the day, month, or year parameter to the `controllers/front/archive.php` archive controller, or the id_category parameter to the `controllers/front/category.php` category controller.
+
+## Summary
+
+* **CVE ID**: [CVE-2021-37538](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2021-37538)
+* **Published at**: 2021-08-24
+* **Advisory source**: [sorcery.ie](https://blog.sorcery.ie/posts/smartblog_sqli/)
+* **Vendor**: PrestaShop
+* **Product**: SmartBlog
+* **Impacted release**: < 4.0.6
+* **Product author**: SmartDataSoft
+* **Weakness**: [CWE-89](https://cwe.mitre.org/data/definitions/89.html)
+* **Severity**: critical (9.8)
+
+## Description
+
+In this smartblog module for PrestaShop, sorcery.ie discovered two SQLis.
+
+In controllers/front/archive.php we can see that the day, month and year parameters are passed to the getArchiveResult() function without sanitisation.
+
+## CVSS base metrics
+
+* **Attack vector**: network
+* **Attack complexity**: low
+* **Privilege required**: none
+* **User interaction**: none
+* **Scope**: unchanged
+* **Confidentiality**: high
+* **Integrity**: high
+* **Availability**: high
+
+**Vector string**: [CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:H/I:H/A:H](https://nvd.nist.gov/vuln-metrics/cvss/v3-calculator?vector=AV:N/AC:L/PR:N/UI:N/S:U/C:H/I:H/A:H)
+
+## Possible malicious usage
+
+* Obtain admin access
+* Remove data from the associated PrestaShop
+* Copy/paste data from sensitive tables to FRONT to exposed tokens and unlock admins's ajax scripts
```

Comment:
Copy/paste data from sensitive tables to the FRONT to expose tokens and unlock admins's ajax scripts